Prozess.txt

```diff
@@ -0,0 +1,240 @@
+		-Standartbibliotheken
+		-Programmierschnittstellen
+		-Laufzeitvariablen
+	Basis-Aufgaben:
+		-Lesen und Schreiben von Dateien
+		-Daten über Netzwerk transportieren
+		-Ein und Ausgabegeräte steueren
+		-Daten verwalten
+		-Sortieren und Suchen
+
+Der Prozess_Scheduler:
+	-Steuerprogramm des Bertiebssystems
+		-Regelt die zeitliche Ausführung mehrerer Prozesse in Bertriebsystemen.
+		-Funktioniert wie eine Warteschlange. 
+		-(Soll in der Lage sein Echtzeit-Systeme zu händeln.)
+		-Kann mehrere Prozesse vor Beendung hin und her switchen, um das gleichzeitige Abl-
+		laufen mehrerer Prozesse zu simulieren.(Dabei kann nur ein Prozess gleichzeitg
+		verarbeitet werden.
+
+	-Grobe Unterteilung in:
+		-unterbrechende
+		-nicht unterbrechende Schedulingstrategien.
+
+	-Unterbrechende Scheduler:
+		-teilen dem CPU von vorneherein nur für eine bestimmte Zeitspanne dem Prozess zu und
+		entziehen Ihn dem Prozess wieder.
+
+	-Nicht unterbrechende Scheduler:
+		Lassen den CPU solange für den Prozess laufen, bis der Prozess den CPU wieder freigibt.
+
+
+Der Thread (Ein Strang/Faden):
+	-Eine nebenläufige Ausführungseinheit innerhalb eines Prozesses
+	
+	Kernel_Threads:
+		 Threads, des gleichen Prozesses können verschiedenen Prozessoren zugeordnet sein.
+		
+	
+	Thread-Kontext:
+		Jeder Thread besitzt einen eigenen Kontext:
+			-unabhägiger Registersatz
+				(Registersatz: Die Menge aller Reigster eines Mikroprozessors.)
+			-einen eigenen Stapel (Stack)
+				(Stack: Ein Stapel mit Last in first Out- Prinzip)
+			
+		Registersatz:
+			Die Menge der Register im Mikroprozessor, die durch Programme direkt
+			angesprochen werden können.
+			3 Typen:
+				-Datenregister zur Zwischenpeicherung von Operanden und 
+				 Rechenergebnissen
+				-Adressregister zur Adressirung von Operanden und Befehlen.
+				-Spezialregister für unterschiedliche Anwendungen z.B Programmzähler
+				 oder Stackpointer.
+
+	Typen:
+		-User-Threads:
+			Nicht im Kernel implementiert, sondern in einer separaten
+			Programmbibliothek.
+				-Taskswitching zwischen Userthreads ohne aufwenidge Systemaufrufe
+				 möglich.
+				-Wen ein Task blcokiert sind alle blockiert.
+
+		-Kernel-Threads:
+			Im kernel implementiert
+				-Taskswichting needs Hardware Support.
+				-Wen ein Task blockiert, kann ein anderer übernehmen.
+				
+Init-Prozess:
+	Ist bei UNIX-Systemen der allererste Prozess mit Prozess-ID 1 (PID1)
+	dieser startet alle anderen Prozesse. Welche das sind und in welcher reihenfolge diese gestartet werden,
+	ist in etc/inittab festgelegt.
+	ein kill init kommt dem Herunterfahren des Systems gleich, da mit dessen Beendigung alle anderen Prozesse beendet werden.
+	Diese anderen Prozesse werden beendet da sie Kind-Prozesse des Init-Prozesses sind.
+	Als aller erstes führt der Init-Prozess jedoch einen POST (Power-On-Self-Test) durch,
+	bei diesem wird:
+		-Funktion der CPU
+		-Funktion der RAM
+		-Funktion der Cache
+		-Funktion der I/O-Software
+	geprüft.
```

Comment:
Clarification needed:

1. Is it really the init process that does that test?
2. What if the processor does not work? Who executes the instructions of the init program?